Code.gs

```diff
@@ -267,12 +267,32 @@ function ConvertToCustomEvent(vevent){
   if (icalEvent.startDate.isDate && icalEvent.endDate.isDate)
     event.isAllDay = true;
     
-  event.startTime = icalEvent.startDate.toJSDate();
-  
-  if (icalEvent.endDate == null)
-    event.endTime = new Date(event.startTime.getTime() + defaultDuration * 60 * 1000);
-  else
-    event.endTime = icalEvent.endDate.toJSDate();
+  if (!event.isAllDay && (icalEvent.startDate.tz == null || icalEvent.endDate.tz == null)){
+    Logger.log("Floating Time detected");
+    var targetTZ = CalendarApp.getTimeZone();
+    Logger.log("Adding Event in " + targetTZ);
```

Comment:
In general, I think breaking this section up with empty lines and comments would be helpful. Something like:

```
//Get Google calendar timezone
...Code

//Calculate timezone offset
...Code

//Shift floating event times by Google calendar timezone
...Code
```